internals/search_queries.py

```diff
@@ -23,26 +23,38 @@
 
 from framework import users
 from framework import utils
-from internals.core_models import FeatureEntry
+from internals import core_enums
+from internals.core_models import FeatureEntry, Stage
 from internals import review_models
 
 
 def single_field_query_async(
     field_name: str, operator: str, val: Union[str, int, datetime.datetime],
     limit: int = None) -> Union[list[int], Future]:
   """Create a query for one FeatureEntry field and run it, returning a promise."""
-  field = QUERIABLE_FIELDS.get(field_name.lower())
-  if field is None:
+  # Note: We don't exclude deleted features, that's done by process_query.
+  field_name = field_name.lower()
+  if field_name in QUERIABLE_FIELDS:
+    # It is a query on a field in FeatureEntry.
+    query = FeatureEntry.query()
+    field = QUERIABLE_FIELDS.get(field_name)
+  elif field_name in STAGE_QUERIABLE_FIELDS:
+    # It is a query on a field in Stage.
+    query = Stage.query()
+    stage_type_dict = STAGE_TYPES_BY_QUERY_FIELD[field_name]
+    # Only consider the appropriate stage type(s).
+    stage_types = [st for st in stage_type_dict.values() if st is not None]
+    query = query.filter(Stage.stage_type.IN(stage_types))
+    field = STAGE_QUERIABLE_FIELDS.get(field_name)
```

Comment:
```suggestion
  if field_name in QUERIABLE_FIELDS:
    # It is a query on a field in FeatureEntry.
    query = FeatureEntry.query()
    field = QUERIABLE_FIELDS[field_name]
  elif field_name in STAGE_QUERIABLE_FIELDS:
    # It is a query on a field in Stage.
    query = Stage.query()
    stage_type_dict = STAGE_TYPES_BY_QUERY_FIELD[field_name]
    # Only consider the appropriate stage type(s).
    stage_types = [st for st in stage_type_dict.values() if st is not None]
    query = query.filter(Stage.stage_type.IN(stage_types))
    field = STAGE_QUERIABLE_FIELDS[field_name]
```

This should fix the mypy errors, and should be safe based on the if statement check.